[ADD] New module: web_widget_date_format

LeartS opened the pull request:
This module adds a new widget for date fields that allows to set a custom
render format using MomentJS tokens

For now it supports only date columns in list views.

LeartS committed:
[ADD] New module: web_widget_date_format

LeartS committed:
Add support for form fields

hbrunn commented:
any reason to create a completely new widget instead of changing the existing date widget to understand more date formats?

houssine78 commented:
Hi @LeartS , 

Thank you!

I agree with @hbrunn . Can't you just extend the existing date format?

LeartS committed:
Update and fix README

LeartS commented:
lol, I ended up modifying just `formats.js` without any new widget or widget changes at all, so it actually already works for standard date and datetime widgets! I just forgot to update the README.

dreispt commented on web_widget_date_format/static/src/js/formats.js:
```diff
@@ -0,0 +1,26 @@
+    var new_format_value = function(value, descriptor, value_if_empty) {
+        // in list view columns, format is passed in the date_format attribute
+        if (descriptor.date_format) {
+            var v = moment(time.auto_str_to_date(value));
```
Sorry if this is a stupid question: is the moment.js lib already included in standard odoo?

LeartS replied:
Yes it is

dreispt replied:
A stupid question? 😜

LeartS replied:
No, already included 😆

LeartS commented:
@hbrunn, @houssine78, @dreispt so do you approve this PR?

hbrunn commented:
I think the module name and the readme are misleading. But when this is changed, I think it's a great addition

LeartS commented:
I already changed the README, do you have any suggestions for the module name?

hbrunn commented:
@pedrobaeza is our name master. Any suggestions?

tarteo reviewed (commented):
Nice module! My suggestion for the name is web_flexible_date_format (something like that) or maybe it can be part of web_widget_datepicker_options.

LeartS committed:
Rename module to web_customisable_date_format

Makes it clearer that the module doesn't create a new widget type,
but uses additional options on the standard date(time) widgets.

LeartS commented:
Hey @hbrunn @dreispt @tarteo @houssine78 .

I've rebased to current 10.0 and renamed the module to `web_customisable_date_format`, I hope it's better. Do you think it can be merged?

nikul-serpentcs commented on web_customisable_date_format/__manifest__.py:
```diff
@@ -0,0 +1,18 @@
+    'category': 'Web',
+    'author': 'MONK Software',
+    'summary': 'Date field widget with custom',
+    'website': 'https://www.monksoftware.it',
```
@LeartS Replace URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10)

nikul-serpentcs commented on web_customisable_date_format/__manifest__.py:
```diff
@@ -0,0 +1,18 @@
+    'version': '10.0.1.0.0',
+    'license': 'AGPL-3',
+    'category': 'Web',
+    'author': 'MONK Software',
```
and also [here](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

simahawk replied:
OCA

nikul-serpentcs reviewed (commented):
Small Change

tarteo commented:
@nikul-serpentcs I think to update the README now is not needed as the PR is already more than one years old. :fearful: Unlike the the author I would like to see OCA to be added.

simahawk commented on web_customisable_date_format/static/src/js/formats.js:
```diff
@@ -0,0 +1,26 @@
+    var original_format_value = formats.format_value;
+    var new_format_value = function(value, descriptor, value_if_empty) {
+        // in list view columns, format is passed in the date_format attribute
+        if (descriptor.date_format) {
```
I'd rewrite it like this
```js
var options = descriptor.options || {},
    date_fmt = descriptor.date_format || options.date_format;
if (date_fmt) {
    return moment(time.auto_str_to_date(value)).format(date_fmt);
}

simahawk commented on web_customisable_date_format/__init__.py:
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
+# © 2016- Leonardo Donelli (donelli at monksoftware it)
```
pls use "Copyright" (not really required in `__init__` BTW)

simahawk commented on web_customisable_date_format/README.rst:
```diff
@@ -0,0 +1,86 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
should be png, but it would be better to switch to new readme format https://github.com/OCA/maintainer-tools/tree/master/template/module/readme :wink:

simahawk commented:
@LeartS thanks for your work :)
PR name says "web_widget_date_format" but module name is "web_widget_date_format".
Please: rename title accordingly, check the comments and the build. Thanks!
